### ISSUE TITLE
Example app

### ISSUE THREAD

**Issue by Nerian** [code-and-effect/effective_datatables#90]
Hi,

I think an example app would be useful for effective_datatables. In my case, I am sometimes not sure if a thing is not working because of some interaction with some other thing from my own app, or because I didn't configure things correctly. I am facing this issue debugging things while I upgrade all the things. It is hard to know where things are broken with so many moving parts.

An example app with a minimal setup will help.

So there it is:

https://github.com/Nerian/effective-datatables-example

It uses sqlite, and it has a table with a filter.

<img width="1341" alt="captura de pantalla 2018-10-09 a las 9 31 20" src="https://user-images.githubusercontent.com/259568/46653360-1f9a0f80-cba6-11e8-9927-ac3d52800117.png">

But I have run in some issues already :)  Which is great, maybe fixing these will fix my issues in CokuApp.

Issue 1
```
Uncaught TypeError: Cannot set property 'confirm' of undefined
    at base.self-c2d8c735753f2da021e5c39a2dd9c5049f097605f468669302c55400e286a3bc.js?body=1:57
    at base.self-c2d8c735753f2da021e5c39a2dd9c5049f097605f468669302c55400e286a3bc.js?body=1:72
```

I get that error on the JS console. It's coming from the effective-bootstrap initialiser.

<img width="1387" alt="captura de pantalla 2018-10-09 a las 9 32 37" src="https://user-images.githubusercontent.com/259568/46653415-47897300-cba6-11e8-86ba-3889bef81987.png">

Issue 2

When I use the filter or column search box and type something a request is sent, but data on the table is not updated.

````
::1 - - [09/Oct/2018:09:30:55 CEST] "POST /datatables/posts_datatable-660041327703.json HTTP/1.1" 200 171
http://localhost:3000/posts -> /datatables/posts_datatable-660041327703.json
Started POST "/datatables/posts_datatable-660041327703.json" for ::1 at 2018-10-09 09:30:55 +0200
Processing by Effective::DatatablesController#show as JSON
  Parameters: {"draw"=>"5", "columns"=>{"0"=>{"data"=>"0", "name"=>"title", "searchable"=>"true", "orderable"=>"true", "search"=>{"value"=>"5", "regex"=>"false"}, "visible"=>"true"}, "1"=>{"data"=>"1", "name"=>"year", "searchable"=>"true", "orderable"=>"true", "search"=>{"value"=>"", "regex"=>"false"}, "visible"=>"true"}}, "order"=>{"0"=>{"column"=>"0", "dir"=>"asc"}}, "start"=>"0", "length"=>"25", "search"=>{"value"=>"", "regex"=>"false"}, "cookie"=>"660041327703", "scope"=>"", "filter"=>{"year"=>"2010"}, "id"=>"posts_datatable-660041327703"}
Can't verify CSRF token authenticity.
   (0.2ms)  SELECT "posts"."year" FROM "posts"
  ↳ app/datatables/posts_datatable.rb:10
   (0.2ms)  SELECT COUNT(*) FROM "posts" WHERE "posts"."year" = ?  [["year", 2000]]
  ↳ /Users/Nerian/.rvm/gems/ruby-2.5.1/gems/activerecord-5.2.1/lib/active_record/log_subscriber.rb:98
  CACHE  (0.0ms)  SELECT COUNT(*) FROM "posts" WHERE "posts"."year" = ?  [["year", 2000]]
  ↳ /Users/Nerian/.rvm/gems/ruby-2.5.1/gems/activerecord-5.2.1/lib/active_record/log_subscriber.rb:98
COLUMN TOOL: order_column title asc "posts"."title"
  Post Load (0.3ms)  SELECT  "posts".* FROM "posts" WHERE "posts"."year" = ? ORDER BY "posts"."title"='' ASC, "posts"."title" ASC LIMIT ? OFFSET ?  [["year", 2000], ["LIMIT", 25], ["OFFSET", 0]]
  ↳ /Users/Nerian/.rvm/gems/ruby-2.5.1/gems/activerecord-5.2.1/lib/active_record/log_subscriber.rb:98
Completed 200 OK in 14ms (Views: 0.3ms | ActiveRecord: 0.7ms)


::1 - - [09/Oct/2018:09:30:55 CEST] "POST /datatables/posts_datatable-660041327703.json HTTP/1.1" 200 171
http://localhost:3000/posts -> /datatables/posts_datatable-660041327703.json
````

I wonder if the `Can't verify CSRF token authenticity.` is the culprit.

Issue 3

When typing on the search box I was getting a datatable exception. Looking at the logs, it was a `Can't verify CSRF token authenticity`. Should I be getting this exception? 

I added this bit to ApplicationController silence it:

`protect_from_forgery with: :null_session, if: Proc.new { |c| c.request.format == 'application/json' }`

But it probably is not fixing anything, just hiding. It may be the cause for issue 2.

I also have `config.authorization_method = true` in both `EffectiveDatatables` and `EffectiveBootstrap` initialisers.

**Comment by Nerian** [comment]
I figured out the issues:

I had set up the order of JS requires incorrectly.

This is the correct order_
```
//= require jquery3
//= require activestorage
//= require turbolinks
//= require rails-ujs

//= require popper
//= require bootstrap-sprockets

//= require effective_datatables
//= require effective_bootstrap
```

That fixes the `confirm` error and the CSRF.

I also figured out why my filters were not working on my app – not this example one.

The filters problem was more tricky. It turns out that Rails new `form_with` does not generate `id` on inputs any more, by default. EffectiveDatatables filters are created by EffectiveBootstrap, which in turns depends on Rails's `form_with`. So the filters' values were not being sent, because there was no id. 

In order to fix this I had to add `config.action_view.form_with_generates_ids = true` to my application config file. Or simply `config.load_defaults 5.2` which takes care of it.
